### DISCUSSION TITLE
Registration Form - Unauthenticated users

### DISCUSSION THREAD

**Discussion by mkashif14** [oqtane/oqtane.framework#4196] [Q&A]
Hi,

We are using Oqtane to build a web-application/portal that would be used for 'Visitor Registration Request'. The registration request page is supposed to be opened by a visitor who does not need to login (in fact no login credentials would be provided to the visitors).

The development for the registration form/page is almost complete (the form looks like below). 
![image](https://github.com/oqtane/oqtane.framework/assets/34096765/c4efce36-a0c3-4143-b493-6a83bb9e861b)

Now we would like to meet the following requirements:
1. Show the "New Registration" page (opening the edit.razor component in "Add" mode) to unauthenticated users (Visitors) such that all the API calls needed to save the form are also functional when the form is submitted by a visitor.
2. Hide the top menu (including the Login button and the logged-in user name field) for this page when accessed by Visitors.

Please let me know how could we achieve the above functionality.

I looked at the following discussion https://github.com/oqtane/oqtane.framework/discussions/1724 which might help us hide the login button, but I believe, if we hide the menu altogether for visitors, that should automatically hide the login button.

**Comment by sbwalker** [comment]
@mkashif14 it sounds like you are building a site that has a specific page where you want visitors to "register" - however this is not a standard registration process for becoming a registered user, it is a registration process for some other purpose (most likely a marketing "contact form" based on the screen shot you provided). You also mentioned that on the page where the registration form is displayed you do not want the site navigation, login, etc... to be visible.

So you will need to create a Register page on your site (most likely a page where Navigation is disabled so that it is a "hidden" page which can only be accessed by browser to its Url). The Register page permissions would allow "All Users" to View the page. You would include an instance of your custom registration module on this page. The Index component of your custom module would include the fields from your screenshot. Oqtane assigns a unique identifier to each site visitor which can be accessed using PageState.VisitorId (note that internally this is based on a cookie so it is specific to a device which means that if a visitor uses multiple devices to access your site they will have multiple VisitorIds).

In terms of storing the information entered by a visitor, you have multiple options. You would obviously use the VisitorId described above as the key. You could create your own custom model, migrations, repository, etc... in your custom module to store the specific information for your registration form. Or you could use the Visitor methods in the SettingService (ie. GetVisitorSettingsAsync and UpdateVisitorSettingsAsync) to store name/value pairs in a generic manner for each visitor. It is up to you based on what you want to do with the information later.

In order to include custom theme behavior, you should create a custom theme. You can easily scaffold a new theme project by using the Create Theme option in Theme Management. In the default.razor component in the theme you would include logic to only show specific components if the user was authenticated:

```
@if (PageState.User != null)
{
  <Menu Orientation="Horizontal" />
}
```

Hopefully this is helpful.
